vcxagency-node/src/index.js

```diff
@@ -15,91 +15,75 @@
-    global.ecsTaskMetadata = await fetchEcsTaskMetadata(appConfig.ECS_CONTAINER_METADATA_URI_V4)
-
-    logger.debug('Going to try reloading trusted certificates.')
-    reloadTrustedCerts(logger)
+async function setGlobalVariables (appConfig) {
+  setLoggingGlobalSettings(appConfig)
+  global.WEBHOOK_RESPONSE_TIMEOUT_MS = appConfig.WEBHOOK_RESPONSE_TIMEOUT_MS
+  const { fetchEcsTaskMetadata } = require('./scripts/fetch-ecs-task-metadata')
+  global.ecsTaskMetadata = await fetchEcsTaskMetadata(appConfig.ECS_CONTAINER_METADATA_URI_V4)
+}
 
-    logger.info(`Loaded effective config: ${stringifyAndHideSensitive(appConfig)}`)
+async function run () {
+  const envConfig = await loadConfiguration()
+  const appConfig = await validateAppConfig(envConfig)
+  await setGlobalVariables(appConfig)
+  // only after global variables are set we can import files which work with logger
+  const logger = require('./logging/logger-builder')(__filename)
+  logger.info(`Starting up with effective config: ${stringifyAndHideSensitive(appConfig)}`)
 
-    logger.debug('Going to fetch certificates/keys to serve.')
-    await fetchCertsFromS3(appConfig, logger)
+  const { startServer } = require('./execution/run')
 
-    logger.debug('Going to build application internals.')
-    const application = await buildApplication(appConfig)
+  process.on('exit', code => {
+    logger.warn(`Process exiting with code: ${code}`)
+  })
 
-    const expressApp = express()
-    expressApp.set('app-name', 'vcxs-api')
+  process.on('SIGTERM', signal => {
+    logger.warn(`Process ${process.pid} received a SIGTERM signal.`)
+    process.exit(0)
+  })
 
-    const httpServer = createWebServer(expressApp, appConfig.SERVER_ENABLE_TLS, appConfig.CERTIFICATE_PATH, appConfig.CERTIFICATE_KEY_PATH, logger)
-    await setupExpressApp(expressApp, application, appConfig)
+  process.on('SIGINT', signal => {
+    logger.warn(`Process ${process.pid} has been interrupted.`)
+    process.exit(0)
+  })
 
-    const port = appConfig.SERVER_PORT
-    logger.debug(`Going to listen on port ${port}`)
-    httpServer.listen(port, () => logger.info(`------------ Listening on port ${port} ------------`))
-  } catch (err) {
-    logger.error(`Unhandled error. Application will be terminated. ${err.stack}`)
-    process.exit(255)
+  function reloadTrustedCerts () {
+    const TRUSTED_CA_CERTS_PATH = '/etc/ssl/certs/ca-certificates.crt'
+    if (fs.existsSync(TRUSTED_CA_CERTS_PATH)) {
+      https.globalAgent.options.ca = fs.readFileSync(TRUSTED_CA_CERTS_PATH)
+      logger.warn(`Loaded additional trusted CA certificates from ${TRUSTED_CA_CERTS_PATH}`)
+    } else {
+      logger.warn('No additional trusted CA certificates were loaded.')
+    }
   }
-}
-
-async function run () {
-  const appConfig = await loadConfiguration()
-  return startApp(appConfig)
+  reloadTrustedCerts()
+  await startServer(appConfig)
+  return false
 }
 
 run()
+  .then((shouldExit) => {
+    if (shouldExit) {
```

Comment:
Does `run()` return `true` anywhere?

Reply:
No so far for agency.

Reply:
What is the point in unreachable code? :P